packages/middleware-sdk-s3/src/throw-200-exceptions.spec.ts

```diff
@@ -15,6 +15,54 @@ describe("throw200ExceptionsMiddlewareOptions", () => {
     jest.clearAllMocks();
   });
 
+  describe("exceptions for code < 200 and >= 300", () => {
+    mockStreamCollector.mockResolvedValue(Buffer.from(""));
+    mockUtf8Encoder.mockReturnValue("");
+    it("throws an exception if code is less than 200", async () => {
+      mockNextHandler.mockReturnValue({
+        response: new HttpResponse({
+          statusCode: 199,
+          headers: {},
+          body: "",
+        }),
+      });
+      const handler = throw200ExceptionsMiddleware(mockConfig)(mockNextHandler, {} as any);
+      try {
+        await handler({
+          input: {},
+          request: new HttpRequest({
+            hostname: "s3.us-east-1.amazonaws.com",
+          }),
+        });
+      } catch (e) {
+        expect(e).toBeDefined();
+        expect(e.name).toEqual("InternalError");
+        expect(e.message).toEqual("S3 aborted request");
+      }
+    });
```

Comment:
The middleware shouldn't throw an error when status code is <200 or >=300.

This code should be:
```suggestion
    const result = await handler({
        input: {},
        request: new HttpRequest({
          hostname: "s3.us-east-1.amazonaws.com",
        }),
      });
    expect(result.response).toBe(mockResponse);
```

Where:

```js
      mockNextHandler.mockReturnValue({
        response: mockResponse,
      });
```